DXRModules/DeusEx/Classes/DXRLoadouts.uc

```diff
@@ -696,53 +697,23 @@ function SpawnItems()
         reducer.Timer();
 }
 
-static function bool IsRefused(class<Inventory> type, optional out int strIdx)
+static function int GetLootAction(class<Inventory> itemClass, optional DataStorage storage)
 {
-    local DataStorage datastorage;
-    local string refusals;
-
-    datastorage = class'DataStorage'.static.GetObj(class'DXRando'.default.dxr);
-    refusals = datastorage.GetConfigKey("item_refusals");
-    if(refusals == "") {
-        return false;
-    }
-    strIdx = InStr(refusals, "," $ type.name $ ",");
-    return strIdx != -1;
+    if (storage == None)
+        storage = class'DataStorage'.static.GetObj(class'DXRando'.default.dxr);
+    return Int(storage.GetConfigKey("LootAction_" $ itemClass.name));
 }
 
-static function UnsetRefuseItem(class<Inventory> type)
+static function SetLootAction(class<Inventory> itemClass, int action, optional DataStorage storage)
 {
-    local DataStorage datastorage;
-    local string refusals, leftPart, rightPart;
-    local int strIdx;
-
-    if (!IsRefused(type, strIdx))
-        return;
-
-    datastorage = class'DataStorage'.static.GetObj(class'DXRando'.default.dxr);
-    refusals = datastorage.GetConfigKey("item_refusals");
-    leftPart = Left(refusals, strIdx);
-    rightPart = Right(refusals, Len(refusals) - (strIdx + Len(type.name) + 1));
-    refusals = leftPart $ rightPart;
-
-    datastorage.SetConfig("item_refusals", refusals, 3600*24*366);
-}
-
-static function SetRefuseItem(class<Inventory> type)
-{
-    local DataStorage datastorage;
-    local string refusals;
-
-    if (IsRefused(type))
-        return;
-
-    datastorage = class'DataStorage'.static.GetObj(class'DXRando'.default.dxr);
-    refusals = datastorage.GetConfigKey("item_refusals");
-
-    if (refusals == "") refusals = ",";
-    refusals = refusals $ type.name $ ",";
+    if (storage == None) {
+        storage = class'DataStorage'.static.GetObj(class'DXRando'.default.dxr);
+    }
 
-    datastorage.SetConfig("item_refusals", refusals, 3600*24*366);
+    if (GetLootAction(itemClass, storage) != action) {
+        log("Setting " $ itemClass.name $ " loot action to: " $ action);
+        storage.SetConfig("LootAction_" $ itemClass.name, action, 3600*24*366);
```

Comment:
I don't like bloating the number of data storage keys per playthrough, we've already had issues with it before since it's a very simple cuckoo hash table

we can do this all in 1 key like before